lib/resty/signal.lua

```diff
@@ -32,6 +32,10 @@ do
         local i = 1
 
         for k, _ in string_gmatch(cpath, "[^;]+") do
+            if k == "?.so" then
+                k = "./"
+            end
+
             local fpath = string_match(k, "(.*/)")
```

Comment:
@zhuizhuhaomeng Do we really want to treat _all_ segments that don't match the pattern as "?.so"? 

This would mean that if a cpath segment is something totally invalid (e.g. "xyz") we would still attempt to load the lib from `./` even though the user has not explicitly told us to.